Automattic/remote-data-blocks
# github markdown block link fixes

# Description

> [!IMPORTANT]  
> Depends on PR #198 

This PR fixes the broken internal links to other markdown docs when using "GitHub File As HTML" block.

## Changes

- Remove unused patterns for "GitHub File As HTML" block
- Allow slashes in overrides in url path when there is just one input variable. Earlier the overrides were delimited by slashes (`/`). For example in url `page_slug/$first/$second` `$first` and `$second` are assigned to first and second input variables respectively. This was a blocker in using the GitHub Doc Markdown file path as url overrides since the file path contains slashes. Now if there is just one input variable, all the url overrides are treated as just one input variable. For example `page_slug/$dir/$path/$file` will assign `$dir/$path/$file` to the input variable.
- If input variables are missing file extension then its added to allow having url overrides without the `.md` suffix
- Fix the internal links in the GitHub File As HTML block by replacing all the href attributes with local links ending in `.md`.


# Testing

- Start the local server using by following the instructions in the [Docs](https://github.com/Automattic/remote-data-blocks/blob/trunk/docs/local-development.md).
- Navigate to the page created in "GitHub File As HTML" example at the url `http://localhost:8888/gh`.
- Edit the page to add a "GitHub File As HTML" block and set the url overrides to `/gh/${file_path}`.
- Pass the file path of the markdown file in url like `http://localhost:8888/gh/${file_path}` (e.g. `http://localhost:8888/gh/docs/workflows/google-sheets-with-code`) and verify that the docs are rendered correctly and the internal links are working.
